### PR TITLE
RELENG-11 - Update release checklist to follow the new process introd…

### PR DESCRIPTION
**Pull request by JohanLorenzo**
…uced by shipit

Let me know if the new process on mergeday looks clearer notably in terms of ownership, @csadilek, @rvandermeulen 🙂 

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

**Review by rvandermeulen — APPROVED** [review]
FYI, after the new .0 release is tagged in GitHub, we'll need to manually edit the release to add the release notes/changelog/milestone information to it that was previously being added upon creation. That's already documented in the Release Management docs, however, so I think this is OK for now. Longer-term, I think we're planning to submit a PR to just refer all Beta release branch creation to our docs anyway.

**Comment by JohanLorenzo** [comment]
Thanks for calling that out! I believe we can automate the release notes/changelog/milestone information in the future. It should be a matter of enhancing [githubscript](https://github.com/mozilla-releng/scriptworker-scripts/tree/83a781f44458f30eee21df34a8d0c9265c686ef5/githubscript) by letting it upload a [`body`](https://github3.readthedocs.io/en/latest/api-reference/repos.html#github3.repos.repo.Repository.create_release). It shouldn't be too much effort 🙂

File: docs/contribute/release_checklist.md [hunk 1]
```diff
@@ -6,59 +6,43 @@ permalink: /contributing/release-checklist
...
-## Starting the next Nightly development cycle
+**Please handle this part once Release Management gives you the go.**
 
+0. Wait for greenlight coming from Release Management.
 1. Create a local development branch off of the `main` branch.
-2. Create a commit named `Start [nightly_version].0.0 development cycle` In [.buildconfig.yml](https://github.com/mozilla-mobile/android-components/blob/main/.buildconfig.yml) and [version.txt](https://github.com/mozilla-mobile/android-components/blob/main/version.txt), bump the current `componentsVersion` up by 1 and ensure this change is also synced with `version.txt`. `[nightly_version]` should follow the Firefox Nightly release number. See [Firefox Release Calendar](https://wiki.mozilla.org/Release_Management/Calendar).
+2. Create a commit named `Start [nightly_version].0.0 development cycle` in [version.txt](https://github.com/mozilla-mobile/android-components/blob/main/version.txt), bump the current `componentsVersion` up by 1 and ensure this change is also synced with `version.txt`. `[nightly_version]` should follow the Firefox Nightly release number. See [Firefox Release Calendar](https://wiki.mozilla.org/Release_Management/Calendar).
```
Comment (csadilek) [inline docs/contribute/release_checklist.md:45]: This ^^ and the code snippet below are still referring to `componentsVersion` in `.buildconfig.yml` which doesn't exist anymore. Just need to update `version.txt` now, right?

**Review by csadilek — CHANGES_REQUESTED** [review]
Found a left-over :)

File: docs/contribute/release_checklist.md (hunk 1 above)
Comment (JohanLorenzo) [reply docs/contribute/release_checklist.md]: Oops, you're totally right! Fixed.

**Pull request merged**